src/sundials_matrix_wrapper.cpp

```diff
@@ -174,6 +174,12 @@ void SUNMatrixWrapper::reset() {
         SUNMatZero(matrix);
 }
 
+void SUNMatrixWrapper::scale(realtype a) {
+    if (matrix)
+        for (int i = 0; i < static_cast<int>(nonzeros()); ++i)
```

Comment:
Move `static_cast<int>(nonzeros())` above loop to call only once and not for each element.